pystac_client/collection_client.py

```diff
@@ -59,7 +77,7 @@ def from_dict(
         cls,
         d: Dict[str, Any],
         href: Optional[str] = None,
-        root: Optional[pystac.Catalog] = None,
+        root: Optional[Union[pystac.Catalog, "Client"]] = None,
```

Comment:
```suggestion
        root: Optional[Union[pystac.Catalog, Client]] = None,
```

If we `from __future__ import annotations` we can simplify this.